images/cassandra-medusa/tests/main.tf

```diff
@@ -0,0 +1,73 @@
+terraform {
+  required_providers {
+    oci = { source = "chainguard-dev/oci" }
+  }
+}
+
+variable "digest" {
+  description = "The image digest to run tests over."
+}
+
+resource "random_pet" "suffix" {
+  length = 1
+}
+
+data "oci_string" "ref" { input = var.digest }
+
+resource "helm_release" "cert-manager" {
+  name             = "cert-manager-${random_pet.suffix.id}"
+  namespace        = "cert-manager"
+  repository       = "https://charts.jetstack.io"
+  chart            = "cert-manager"
+  create_namespace = true
+
+  set {
+    name  = "installCRDs"
+    value = "true"
+  }
+}
+
```

Comment:
This is looking great so far. I think we could further extend the tests here by doing some log validation after deploying via helm.

Some examples from our grafana image:

- https://github.com/chainguard-images/images/blob/03632b6ed8e41ee3fd121eca2dec96733c0f10d8/images/grafana/tests/main.tf#L33
- https://github.com/chainguard-images/images/blob/03632b6ed8e41ee3fd121eca2dec96733c0f10d8/images/grafana/tests/smoke-test.sh

Reply:
Thanks @mamccorm , I guess these logs based test might not gave great results as its a CRD based tests and we might need to create objects related to medusa from CRD, only then we can verify it to that level. For that, we will have to use AWS and probably will need a S3 bucket, for ref: https://docs.k8ssandra.io/tasks/backup-restore/

Reply:
Thanks, yea looks like the operator only creates a pod for medusa whenever it need to run a backup job.

I was wondering can we at least force the operator to create the pod in the test, even if the backup itself fails.

i.e:

```bash
apiVersion: k8ssandra.io/v1
kind: K8ssandraCluster
metadata:
  name: my-k8ssandra-cluster
  namespace: cassandra-medusa
spec:
  cassandra:
    cluster: my-cassandra-cluster
    datacenters:
    - metadata:
        name: dc1
      size: 3
```

But I am not really sure how to deploy this. looks like its looking for CRDs now